masonry/src/event_loop_runner.rs

```diff
@@ -280,12 +279,18 @@ impl MasonryState<'_> {
                 .unwrap();
                 let scale_factor = window.scale_factor();
                 self.window = WindowState::Rendering {
-                    window,
+                    window: window.clone(),
                     surface,
                     accesskit_adapter: adapter,
                 };
                 self.render_root
                     .handle_window_event(WindowEvent::Rescale(scale_factor));
+                // Render one frame before showing the window to avoid flashing
+                if visible {
+                    let (scene, _tree_update) = self.render_root.redraw();
```

Comment:
Discarding this tree update feels likely to be very bad.

Reply:
Yeah, ignoring the tree update could lead to issues. I’ve updated the code to include it, following the approach used in handling the `RedrawRequested` event. Let me know if it looks good now.